src/node/internal/internal_http_outgoing.ts

```diff
@@ -916,11 +921,21 @@ export class OutgoingMessage extends Writable implements _OutgoingMessage {
     // exists and is currently writable, it would flush any pending data to the socket and then
     // write the current chunk's data directly into the socket. Afterwards, it would return with the
     // value returned from socket.write().
-    if (this.#buffer != null) {
+    if (this.#buffer != null && (this._header !== null || this._headerSent)) {
```

Comment:
I know this already merged but as a follow up, the next time someone is in this code, since the `_header !== null || this._headerSent` check is duplicated several times it might make sense to separate out into a utility function for reuse.